v1/services/storage/store.go

```diff
@@ -208,7 +208,11 @@ func (s *Store) ReadGroup(ctx context.Context, req *datatypes.ReadGroupRequest)
 		return nil, err
 	}
 
-	shardIDs, err := s.findShardIDs(database, rp, false, start, end)
+	// Due to some optimizations around how flux's `last()` function is implemented with the
+	// storage engine, we need to detect if the read request requires a descending
+	// cursor or not.
+	descending := !reads.IsAscendingGroupAggregate(req)
+	shardIDs, err := s.findShardIDs(database, rp, descending, start, end)
```

Comment:
The `last()` bug strikes again!

Reply:
@danxmoran is there something we need to change?

Reply:
No, was just surprised to see this bug pop up in a 2nd place.